### PR TITLE
[Gutenberg] Send http status code to network error callback.

### PR DESCRIPTION
**Pull request by etoledom**
`gutenberg-mobile` side PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1686

This PR ensures that the real http status code is returned on the network error callback.

To test:
- Run metro server on https://github.com/wordpress-mobile/gutenberg-mobile/pull/1686
- Check test steps on https://github.com/wordpress-mobile/gutenberg-mobile/pull/1686

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergNetworking.swift [hunk 1]
```diff
@@ -24,13 +24,13 @@ struct GutenbergNetworkRequest {
...
-        }, failure: { (error, HTTPResponse) in
-            completion(.failure(error))
+        }, failure: { (error, httpResponse) in
+            completion(.failure(error.nsError(with: httpResponse)))
         })
     }
 
     private func dotComPath(with dotComID: NSNumber) -> String {
-        return path.replacingOccurrences(of: "/wp/v2/", with: "/wp/v2/sites/\(dotComID)/")
+        return path.replacingOccurrences(of: "/wp/v2/", with: "/wp/v2/sitesa/\(dotComID)/")
```
Comment (SergioEstevao) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergNetworking.swift]: Is this correct: `sitesa` ?

Comment (etoledom) [reply WordPress/Classes/ViewRelated/Gutenberg/GutenbergNetworking.swift]: Oh no! I changed that to test an error code. Didn't change it back 😬
Fixed now!

**Comment by etoledom** [comment]
> I'm unable to see the image being loaded again after I come back online.

This was probably because of the typo introduced in the path. Sorry about that!
It's ready now for another look 👍

**Comment by loremattei** [comment]
Hey! I moved this to 14.1 because 14.0 has been cut. Please, ping me if you want it to make it to 14.0 😀

**Comment by loremattei** [comment]
Hey! I moved this to 14.2 because 14.1 has been cut. Please, ping me if you want it to make it to 14.1 😀

**Pull request merged**